### DISCUSSION TITLE
GitHub Copilot suddenly generating gibberish suggestions

### DISCUSSION THREAD

**Discussion by danbrakeley** [community/community#72633] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I've been using GitHub Copilot with VS Code for a few months now without issue, but I sat down to do some work this morning (7:00am EST), and all the suggestions it now makes are gibberish. It was working fine as of 4:00pm (EST) last night.

Here are a few examples:

![successsuccesssuc](https://github.com/community/community/assets/6772870/6f2cf094-790e-4640-9417-9add4656da8b)
![pullsullullsul](https://github.com/community/community/assets/6772870/09530352-0762-488d-b398-875c50fb2168)
![brackets](https://github.com/community/community/assets/6772870/2280f809-676a-49df-b6f0-aa475ff16ece)
![githubthegithubgithubgithubgithubgithubfro](https://github.com/community/community/assets/6772870/f7ed2c8d-4eef-4baf-bb04-cc4463a3501a)
![modelslashslashslashslashslashslashclosebracket](https://github.com/community/community/assets/6772870/30687346-ebf9-49ef-aca5-50eb97e456f5)

I'm running Windows 10 Pro 22H2 (19045.3570), VS Code 1.83.1 (user setup), and GitHub Copilot extension v1.130.0. As far as I know, nothing changed on my end between last night and this morning, although if I'm reading this right, a new version of the Copilot VS Code extension was released late last night ("Last released: 2023-10-25, 20:44:40").

Earlier this morning I did try uninstalling the extension, closing all instances of Code (double checking in the Task Manager that no code.exe processes were left), manually deleting the folders matching `~/.vscode/extensions/github.copilot*`, then re-running Code and re-installing the extension. This did not make any noticeable change to the gibberish suggestions being generated.

**[EDIT]** I tried rolling back the extension to 1.290.0, but it continues to make the same nonsense suggestions.

**Comment by kiliss** [comment]
same problem

**Comment by christian-gama** [comment]
I got the same problem, but now it stopped working completely - no suggestions are being made.

**Reply by SpencerWittrock** [reply]
Same thing here

**Comment by agustin-vieta** [comment]
Same problem here! -_-

**Comment by gregorysouzasilva** [comment]
same problem here. Started yesterday night for me after updating. Moved back to v1.130. It's giving some right suggestion but pretty slow. :/

**Comment by jbFrameworks** [comment]
Same here; results in nonsensical output with many repeating words and symbols. For me, I first noticed the issue yesterday after applying a prompted extension update (to v1.130.0).

TL;DR/FYI: Having researched this issue via Internet search and https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-common-issues-with-github-copilot, I turned on Copilot's advanced debugging (via the extension's "github.copilot.advanced" setting) and, now, in VSC's OUTPUT window (filtered for "GitHub Copilot" activity), I see the following occurring repeatedly|regularly:

[ERROR] [fetch] [2023-10-26T18:18:26.012Z] Unhandled status from server:,504,request timeout

preceded by:

[INFO] [default] [2023-10-26T18:18:15.972Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-10-26T18:18:26.010Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 10037 ms

**Reply by jbFrameworks** [reply]
Ironically, Windows Copilot says... 

"Yes, you are not alone in this issue. It seems like many users are experiencing the same problem with GitHub Copilot generating garbage code or repeating words and symbols. Some possible causes for this issue are:

- GitHub Copilot is still in technical preview and is not perfect. It may sometimes generate incorrect, insecure, or nonsensical code.
- GitHub Copilot is learning from the public code available on GitHub and other sources. It may sometimes pick up bad practices, bugs, or outdated code from these sources.
- GitHub Copilot is influenced by the context of your code and the comments you write. It may sometimes misunderstand your intention or generate code that does not match your logic or style.

The GitHub Copilot team is working hard to improve the quality and reliability of the code suggestions. They are also collecting feedback from the users to identify and fix the issues. You can help them by reporting any problems you encounter using the Share feedback button in the VS Code status bar or by sending an email to copilot-feedback@github.com.

I hope this helps you understand why GitHub Copilot is not working as expected and what you can do to help improve it. If you have any other questions, please let me know. 😊"

**Comment by DonRedwoood** [comment]
Exact same problem here. It's been providing gibberish all day until about an hour ago, when it stopped working completely.

**Comment by IrisesD** [comment]
Same problem here.

**Comment by GabeMachado** [comment]
Same problem here. It started giving completely wrong autocomplete ideas and then it stopped working for good a few hours ago. I've already tried adding vs code to the firewall, rolling back the copilot version, uninstalling both vs code and copilot and I've also tried restarting the PC. Does anyone know how to fix it?

`[ERROR] [fetch] [2023-10-26T20:22:26.759Z] Unhandled status from server:,504,request timeout`

**Reply by James991116** [reply]
Same here

**Comment by danbrakeley** [comment]
I did nothing locally (still running extension v1.130.0), but it seems to be working great so far today. So must have been a backend issue that has been fixed. I'll close this bug report as resolved.
